### ISSUE TITLE
Feature Request: Stage Radio 24/7

### ISSUE THREAD

**Issue by rexjohannes** [littlestudios69/Pika#12]
Please add the Feature that the Bot is 24/7 in Stage Channel and starts Playing an specifc Radio on Join

**Comment by rexjohannes** [comment]
Like this Like this: http://rexum.rf.gd/ZUrbj34hN0o.png from Groovy

**Comment by ImFelixprogram** [comment]
This would be impractical because of multiple reasons:
1. The frequent restarts that occur because of updates & fixes, which leads the bot to leave its' channel and the join command would have to be issued once more
2. Users could not listen to custom music / radio then because a bot can only join one channel
3. That would use too much resources to have radio playing all the time, even without users in that channel

Maybe Tayron has another opinion on that topic but unless that is the case, we will not add that feature in the near future

~ Felixprogram, CTO of Little Studios

**Comment by taynotfound** [comment]
Hey Johannes,


Unfortunately I don't think we can do that because of several Reasons:

1. Our Ressources would go wild
2. I don't think our version of DJS can handle stages
3. And as Felix said you would need a ton of Bots that users can use and we don't want that we basically want that you use our Bot not others.


If you have arguments for it then please state them here.

**Comment by taynotfound** [comment]
And yes we could do multiple bots but that would be Hella in efficient and would suck too much Ressources

**Comment by rexjohannes** [comment]
Hi,

You dont need to play the Music if no one is in the channel just if someone joins. And yes you cannot play music in other channels then but the server owner can choose. And the bot can save all stage Channels in an db and join it on reboot

**Comment by taynotfound** [comment]
We will look into it

**Comment by ImFelixprogram** [comment]
Hi,

there's a couple of problems with your suggestions.
We'll go over the good things first. You're right that we don't have to let the music playing when noone is listening and also about the fact that it should be the server owner's decision, if he / she wants to allow users to listen to their own music.
**But** the really big problem is again performance. First, even if we're not playing anything, we still need to be connected to that channel which, yes, is better than playing music as well, but still not a great solution. And yes, we could let the bot leave and then join when a user joins again _but_ then - and also for the first one to only stop and start the music - we would have to monitor for every join and leave on any channel on any server that has that feature enabled. And there's another problem with handling the problem with restarts by saving the data in a database. We would have to check for every server with that feature enabled multiple conditions at startup, if the channel and server still exist, if the bot still has appropriate permissions to join, if a user is in that channel and of course the join itself which is not only performance-intensive but also could rate-limit our Bot if we wouldn't give it a long delay which would also not be that good.
So, if we can figure out a solution to that problems we will try to implement that feature, it will be on our To-Do at lower priority than bugfixes and code-cleanup though because we want to get the Bot and its' code polished first before we start adding new features again.
We'll close that topic for now, if you have anything to add to this topic, for example a possible solution for those problems, you can of course message Tayron or me directly, also, if you have time and maybe want to try adding those features yourself, you can of course open a pull request at any time.

~ Felixprogram, CTO of Little Studios
